manticore/seth.py

```diff
@@ -1082,71 +1239,22 @@ def global_coverage(self, account_address):
-        seen = self.context['coverage'] #.union( self.context.get('code_data', set()))
-        runtime_bytecode = world.storage[account_address]['code']
 
-        end = None
-        if  ''.join(runtime_bytecode[-44: -34]) =='\x00\xa1\x65\x62\x7a\x7a\x72\x30\x58\x20' \
-            and  ''.join(runtime_bytecode[-2:]) =='\x00\x29':
-            end = -9-33-2  #Size of metadata at the end of most contracts
+        with self.locked_context('runtime_coverage') as coverage:
+            seen = coverage
+        #runtime_bytecode = world.storage[account_address]['code']
+        runtime_bytecode = self.get_metadata(account_address).runtime_bytecode
 
         count, total = 0, 0
-        for i in evm.EVMAsm.disassemble_all(runtime_bytecode[:end]) :
+        for i in evm.EVMAsm.disassemble_all(runtime_bytecode) :
             if (account_address, i.offset) in seen:
                 count += 1            
             total += 1
 
         return count*100.0/total
 
-    def report_coverage(self, account_address):
-        ''' Output a code coverage report for contract account_address '''
-        account_address = int(account_address)
-        #This will just pick one of the running states.
-        #This assumes the code and the accounts are the same in all versions of the world
-        #Search one state in which the account_address exists
-        world=None
-        for state_id in self.all_state_ids:
-            world = self.get_world(state_id)
-            if account_address in world.storage:
-                break
-
-        seen = self.context['coverage'] #.union( self.context.get('code_data', set()))
-        runtime_bytecode = world.storage[account_address]['code']
-        class bcolors:
-            HEADER = '\033[95m'
-            OKBLUE = '\033[94m'
-            OKGREEN = '\033[92m'
-            WARNING = '\033[93m'
-            FAIL = '\033[91m'
-            ENDC = '\033[0m'
-            BOLD = '\033[1m'
-            UNDERLINE = '\033[4m'
-
 
-        end = None
-        if  ''.join(runtime_bytecode[-44: -34]) =='\x00\xa1\x65\x62\x7a\x7a\x72\x30\x58\x20' \
-            and  ''.join(runtime_bytecode[-2:]) =='\x00\x29':
-            end = -9-33-2
-
-
-        output = ''
-        offset = 0
-        count = 0
-        total = 0
-        for i in evm.EVMAsm.disassemble_all(runtime_bytecode[:end]) :
-            
-            if (account_address, offset) in seen:
-                output += bcolors.OKGREEN
-                output += "** 0x%04x %s\n"%(offset, i)
-                output += bcolors.ENDC
-                count += 1
-            else:
-                output += "   0x%04x %s\n"%(offset, i)
-            
-            total += 1
-            offset += i.size
-
-        output += "Total assembler lines: %d\n"% total
-        output += "Total assembler lines visited: %d\n"% count
-        output += "Coverage: %2.2f%%\n"%  (count*100.0/total)
-        return output
+    #Fixme. This is for avoinding  super._did_finish_run_callback
```

Comment:
TODO: find a better way to suppress execution of Manticore._did_finish_run_callback